src/Hubspot/SyncToService.php

```diff
@@ -103,54 +116,155 @@ public function sync(EntityInterface $entity) {
     if (empty($field_mapping)) {
       return;
     }
+
+    // Prepare the paylaod to send to Hubspot.
+    $hubspot_field_properties = $this->preparePayload($field_mapping);
+    if (empty($hubspot_field_properties)) {
+      return;
+    }
+
+    // Now, do the actual syncing depending on the entity type.
+    $function_name = 'sync' . $entity_mapping['type'];
+    $this->$function_name($hubspot_field_properties, $entity_mapping['id']);
+  }
+
+  /**
+   * Prepare the payload for syncing the properties.
+   *
+   * @param array $field_mapping
+   *   The array of fields that should be mapped.
+   *
+   * @return mixed
+   *   An array of Hubspot properties with their values.
+   */
+  protected function preparePayload(array $field_mapping) {
+    $hubspot_field_properties = [];
+    foreach ($field_mapping as $drupal_field_name => $hubspot_field) {
+      if (!$hubspot_field['status']) {
+        return;
+      }
+
+      $hubspot_field_properties[] = [
+        'property' => $hubspot_field['id'],
+        'value' => $hubspot_field['value'],
+      ];
+    }
+
+    return $hubspot_field_properties;
   }
 
   /**
    * Syncs the contact details with Hubspot.
    *
-   * @param \Drupal\Core\Entity\EntityInterface $entity
-   *   The user entity.
+   * @param array $hubspot_field_properties
+   *   An array of Hubspot properties with their values.
+   * @param int $hubspot_entity_id
+   *   The Hubspot contact ID if the contact has already been synced to Hubspot.
+   *
+   * @throws \Exception
    */
-  protected function syncContact(EntityInterface $entity) {
-    // TODO: Create the necessary properties from the entity object.
+  protected function syncContact(array $hubspot_field_properties, $hubspot_entity_id = NULL) {
+    $contacts = new Contacts($this->client);
 
-    // TODO: Check if a contact already exists in Hubspot, if so, we update.
+    // Create the contact if it hasn't been synced yet.
+    if (!$hubspot_entity_id) {
+      $response = $contacts->create($hubspot_field_properties);
+
+      // If we were successful, save the Hubspot contact ID in the entity.
+      if ($response->getStatusCode() == 200) {
+        $body = $response->getBody();
+
+        $this->entity->set('hubspot_contact_id', $body['vid'])->save();
+      }
+    }
+    else {
+      $contacts->update($hubspot_entity_id, $hubspot_field_properties);
+    }
```

Comment:
**CHANGE REQUEST**

We don't know (especially for custom entities) where and if the remote ID is stored. For some entities it may not be stored at all. For example, it will be stored at the User entity but not in the profiles. So let's have these methods (and the `sync` method) return the remote ID (whether created or updated) and it's up to the caller (in this case the queue worker) to decide what to do with it.

Then dispatch an event from the queue worker `commerce_hubspot.entity_sync_to.post_sync` that makes available the Drupal entity and the remote ID and subscribers can react accordingly.